packages/toolpad-app/src/components/EditableTreeItem.tsx

```diff
@@ -96,6 +94,7 @@ export default function EditableTreeItem({
 
   const handleFocus = React.useCallback((event: React.FocusEvent<HTMLInputElement>) => {
     event.target.select();
+    event.stopPropagation();
```

Comment:
Just out of curiosity, what's the reason to stop propagation here?

Reply:
The issue was that the `autoFocus` triggered the `handleFocus` handler, which somehow also caused the `handleBlur` handler to immediately trigger and cause the input to become unfocused, nullifying the effect of clicking the Rename button. Stopping propagation in the focus handler fixed that.